index.js

```diff
@@ -1,254 +1,263 @@
+            dependencies: []
+        })
         currentDefinition = definitions[name]
         return api
     }
 
-    function include(subSystem) {
+    const include = (subSystem) => {
         debug('Including definitions from sub system %s into system %s', subSystem.name, params.name)
-        definitions = assign({}, definitions, subSystem._definitions)
+        definitions = Object.assign({}, definitions, subSystem._definitions)
         return api
     }
 
+    //FIXME: there is some issue with implicit 'this' usage
     function dependsOn() {
         if (!currentDefinition) throw new Error('You must add a component before calling dependsOn')
-        currentDefinition.dependencies = toArray(arguments).reduce(toDependencyDefinitions, currentDefinition.dependencies)
+        currentDefinition.dependencies = [...arguments].reduce(toDependencyDefinitions, currentDefinition.dependencies)
         return api
     }
 
-    function toDependencyDefinitions(accumulator, arg) {
-        var record = typeof arg === 'string' ? { component: arg, destination: arg } : defaults({}, arg, { destination: arg.component })
+    const toDependencyDefinitions = (accumulator, arg) => {
+        const record = typeof arg === 'string' ? {
+            component: arg,
+            destination: arg
+        } : Object.assign({},{destination: arg.component}, arg);
         if (!record.component) throw new Error(format('Component %s has an invalid dependency %s', currentDefinition.name, JSON.stringify(arg)))
-        if (find(currentDefinition.dependencies, { destination: record.destination })) throw new Error(format('Component %s has a duplicate dependency %s', currentDefinition.name, record.destination))
+        if (currentDefinition.dependencies.find(dep => dep.destination === record.destination)) {
+            throw new Error(format('Component %s has a duplicate dependency %s', currentDefinition.name, record.destination))
+        }
         return accumulator.concat(record)
+
     }
 
-    function start(cb) {
+    const start = (cb) => {
         debug('Starting system %s', params.name)
         started = []
-        var p = new Promise(function(resolve, reject) {
-          async.seq(sortComponents, ensureComponents, function(components, cb) {
-            debug('System %s started', params.name)
-            running = components
-            cb(null, components)
-          })(function(err, components) {
-            if (err) return reject(err, {})
-            resolve(components)
-          })
+        const p = new Promise(function (resolve, reject) {
+            async.seq(sortComponents, ensureComponents, (components, cb) => {
+                debug('System %s started', params.name)
+                running = components
+                cb(null, components)
+            })(function (err, components) {
+                if (err) return reject(err, {})
+                resolve(components)
+            })
```

Comment:
Looks like there's some extra indentation

Reply:
I've to review IDE settings because I'm formatting with eslint rules